glide-core/src/client/value_conversion.rs

```diff
@@ -892,6 +879,19 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         }
+        ExpectedReturnType::SingleOrMultiNode(value_type, value_checker) =>  match value {
+            Value::Map(ref map) => match value_checker {
+                    Some(func) => {
+                        if !map.is_empty() && func(map[0].clone().1) {
+                            convert_to_expected_type(value.clone(), Some(ExpectedReturnType::Map { key_type: &None, value_type }))
+                        } else {
+                            Ok(value.clone())
+                        }
+                    }
+                    None => convert_to_expected_type(value.clone(), Some(ExpectedReturnType::Map { key_type: &None, value_type })),
```

Comment:
No need to deep copy the value when you can move it.
```suggestion
                        if !map.is_empty() && func(map[0].clone().1) {
                            convert_to_expected_type(value, Some(ExpectedReturnType::Map { key_type: &None, value_type }))
                        } else {
                            Ok(value)
                        }
                    }
                    None => convert_to_expected_type(value, Some(ExpectedReturnType::Map { key_type: &None, value_type })),
```